code-challenge/src/App.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {useEffect, useState} from 'react';
+import './App.css';
+import ApiService from './services/ApiService';
+import { AppBar, Toolbar, Typography } from "@material-ui/core";
+import ArticleList from "./components/ArticleList/ArticleList.lazy";
+import Search from "./components/Search/Search.lazy";
+import { useDispatch, useSelector } from 'react-redux';
+import { store } from "./redux/createStore";
+
+type RootState = ReturnType<typeof store.getState>;
+
+
+function App() {
+  const [error, setError] = useState(null);
+  const searchTerm = useSelector((state: RootState) => state.searchTerm);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const apiService = new ApiService();
+    apiService.get()
+        .then((response) => {
+            // console.log(response)
+          if (response.error) {
+              setError(response.error);
+          } else {
+              dispatch({ type: 'SET_BASE_ARTICLES', payload: { articles: response } });
+          }
+        });
+  }, []);
+
+  return (
+    <div className="App">
+        <AppBar>
+            <Toolbar>
+                <Typography variant="h6">
+                    New York Times Top Articles
+                </Typography>
+                <Search />
+            </Toolbar>
+        </AppBar>
+        {error && 'The articles could not be retrieved. Please try again later.'}
+        {!error && <ArticleList />}
```

Comment:
I'm happy to see a view for a potential error state

Reply:
i kept getting 429s because I made too many requests initially so this was purely selfish

Reply:
You're not the first person to report that they were throttled when devving the challenge. We might need to consider using a different unauthed API if this causes headaches.

Reply:
it was pure carelessness when i had useEffect in place with a recursive variable initially. I personally think it adds a layer of extra care for gracefully handling errors from APIs if there's a rate limit. I think you should leave it in. Also having a 3rd party API helps show a front-end engineer can work with APIs without them being on the same origin when developing locally.